src/plugins/mongoose/logger/plugin.js

```diff
@@ -241,10 +252,52 @@ export function plugin(mongooseInstance) {
     schema._klLoggerLogger = Logger;
     schema._klLoggerOptions = mOptions;
     schema.pre('save', function preSave(next) {
-      this._klLoggerModifiedPaths = this.modifiedPaths();
-      this._klLoggerIsNew = this.isNew;
+      // Check if this has already been set by another pre hook
+      if (!this._klLoggerStaticContext) {
+        this._klLoggerModifiedPaths = this.modifiedPaths();
+        this._klLoggerIsNew = this.isNew;
+      }
       next();
     });
+    schema.pre('findOneAndUpdate', async function preUpdate(next) {
+      const search = this._conditions || {};
+      const options = this.options || {};
+      const updated = this._update || {};
+
+      const callstack = options._klLoggerSaveCallStack;
+      delete options._klLoggerSaveCallStack;      
+
+      const existing = await this.findOne(search);
+      if (existing) {
+        this._klLoggerInitialDoc = existing.loggableObject();
+      }
+      
+
+      // filter out mongo fields starting with $
+      const keys = Object.keys(updated).filter(x => !x.includes('$'));
+
+      this._klLoggerModifiedPaths = keys;
+      this._klLoggerIsNew = !existing && options.upsert;
+      this._klLoggerSaveCallStack = callstack;
+      
+      next();
+    });
+    schema.post('findOneAndUpdate', async function(result, done) {
+      
+      if (!result) return done();
+      result._klLoggerIsNew = this._klLoggerIsNew;
+      result._klLoggerModifiedPaths = this._klLoggerModifiedPaths;
+      result._klLoggerInitialDoc = this._klLoggerInitialDoc;
+      result._klLoggerActor = this._klLoggerActor;
+      result._klLoggerSaveCallStack = this._klLoggerSaveCallStack;
+
+      // Set flag to make sure that the pre-save doesnt overide this info
+      result._klLoggerStaticContext = true;
+
+      // Skip validation when because findOneAndUpdate does not do model validation.
+      await result.save({ validateBeforeSave: false });
```

Comment:
What is the point of saving the updated document again here? 
Is it just to make `schema.post('save')` hook to fire and log the event? 
I think it would be better to abstract out the event logging logic and reuse it in both hooks.